lib/srv/db/mysql/engine.go

```diff
@@ -178,6 +177,11 @@ func (e *Engine) receiveFromClient(clientConn, serverConn net.Conn, clientErrCh
 	for {
 		packet, err := protocol.ParsePacket(clientConn)
 		if err != nil {
+			if utils.IsOKNetworkError(err) {
+				log.Debug("Client connection closed.")
+				clientErrCh <- nil
```

Comment:
how about a `defer close(clientErrCh)` at the top of this method?

Reply:
Done.